packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -538,6 +514,57 @@ public void close() {
     }
   }
 
+  public float getMaxZoomLevel() {
+    return cameraCharacteristics.get(CameraCharacteristics.SCALER_AVAILABLE_MAX_DIGITAL_ZOOM);
+  }
+
+  public float getMinZoomLevel() {
+    return 1f;
+  }
+
+  public void setZoomLevel(@NonNull final Result result, float zoom) throws CameraAccessException {
+    float maxZoom = this.getMaxZoomLevel();
+    float minZoom = this.getMinZoomLevel();
+
+    if (zoom > maxZoom || zoom < minZoom) {
+      String errorMessage =
+          String.format(
+              Locale.ENGLISH,
+              "Zoom level out of bounds (zoom level should be between %f and %f).",
+              minZoom,
+              maxZoom);
+      result.error("ZOOM_ERROR", errorMessage, null);
+      return;
+    }
+
+    //Zoom area is calculated relative to sensor area (activeRect)
+    Rect zoomCropPreview = getZoomRect(zoom, activeArraySize.width(), activeArraySize.height());
+    if (captureRequestBuilder != null) {
+      captureRequestBuilder.set(CaptureRequest.SCALER_CROP_REGION, zoomCropPreview);
+      cameraCaptureSession.setRepeatingRequest(captureRequestBuilder.build(), null, null);
+    }
+
+    result.success(null);
+  }
+
+  //Calculate zoom area according input image size
+  private static Rect getZoomRect(float zoom, int imgWidth, int imgHeight) {
```

Comment:
This logic probably needs a test

Reply:
I have moved the zoom logic into its own class `CameraZoom` and make sure it is fully tested. To do so I had to bring in Robolectric (to be able to correctly use the `android.graphics.Rect` class in my unit tests). This also required an update of Gradle. I updated Gradle to stay in sync with the current version Flutter is using on stable (Gradle: 3.5.0, Gradle wrapper: 5.6.2).